src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/Dataflow/ReflectionMethodBodyScanner.cs

```diff
@@ -48,30 +51,55 @@ public static bool RequiresReflectionMethodBodyScannerForMethodBody(FlowAnnotati
 
         public static bool RequiresReflectionMethodBodyScannerForAccess(FlowAnnotations flowAnnotations, FieldDesc fieldDefinition)
         {
-            return flowAnnotations.RequiresDataflowAnalysis(fieldDefinition);
+            return flowAnnotations.RequiresDataflowAnalysis(fieldDefinition) ||
+                fieldDefinition.DoesFieldRequires("RequiresUnreferencedCodeAttribute", out _) ||
+                fieldDefinition.DoesFieldRequires("RequiresDynamicCodeAttribute", out _);
+        }
+
+        void CheckAndReportRequires(TypeSystemEntity calledMember, in MessageOrigin origin, string requiresAttributeName)
+        {
+            // If the caller of a method is already marked with `RequiresUnreferencedCodeAttribute` a new warning should not
+            // be produced for the callee.
```

Comment:
This now supports other requires attributes as well - update the comment please.